File: src/main/java/com/github/nsc/de/shake/parser/node/variables/VariableUsageNode.java
package com.github.nsc.de.shake.parser.node.variables;

import com.github.nsc.de.shake.parser.node.IdentifierNode;
import com.github.nsc.de.shake.parser.node.ValuedNode;
import com.github.nsc.de.shake.util.characterinput.position.PositionMap;

public class VariableUsageNode extends ValuedNode {

    private final IdentifierNode variable;

    public VariableUsageNode(PositionMap map, IdentifierNode variable) {
        super(map);
        this.variable = variable;
    }
    
    public IdentifierNode getVariable() {
        return variable;
    }

    @Override
    public String toString() {
        return "{variable="  + this.variable + '}';
    }
}

File: parser/src/main/kotlin/com/github/nsc/de/shake/parser/node/variables/VariableUsageNode.kt
package com.github.nsc.de.shake.parser.node.variables

import com.github.nsc.de.shake.parser.node.ValuedNode
import com.github.nsc.de.shake.parser.node.IdentifierNode
import com.github.nsc.de.shake.util.characterinput.position.PositionMap

class VariableUsageNode(map: PositionMap, val variable: IdentifierNode) : ValuedNode(map) {
    override fun toJson(): Map<String, *> = mapOf("name" to "VariableUsageNode", "variable" to variable.json)
}